### PR TITLE
Minor Updates

### PR DESCRIPTION
**Pull request by polarker**

### CODE REVIEW

File: packages/web3/src/contract/contract.ts [hunk 1]
```diff
@@ -1725,14 +1725,15 @@ function getBytecodeTemplate(
   // For the default TxScript main function
   const numberOfMethods = '01'
   const isPublic = '01'
-  const modifier = usePreapprovedAssets ? '03' : '00'
+  const scriptUseApprovedAssets = attoAlphAmount !== undefined || tokens !== undefined
```
Comment (Lbqds) [inline packages/web3/src/contract/contract.ts:1728]: Question: In what situations does a contract method use preapproved assets, but the script does not use preapproved assets?

**Review by Lbqds — APPROVED** [review]
LGTM, only one question.

File: packages/web3/src/contract/contract.ts (hunk 1 above)
Comment (polarker) [reply packages/web3/src/contract/contract.ts:1728]: That's impossible, but it's possible that the method does not use preapproved assets, but the script use preapproved assets. E.g. the withdraw function of token faucet.

Comment (Lbqds) [reply packages/web3/src/contract/contract.ts:1728]: Got it, thanks for the explanation

**Pull request merged**